source/vistas/core/graphics/mesh.py

```diff
@@ -79,6 +79,13 @@ def __init__(
             glEnableVertexAttribArray(1)
             glVertexAttribPointer(1, 3, GL_FLOAT, GL_FALSE, sizeof(GLfloat) * 3, None)
 
+        if self.has_color_array:
```

Comment:
Yeah, I was going for more of a "decide the color in the fragment array" to offload the work to the shader, rather than texture.